pkg/distributor/ha_tracker.go

```diff
@@ -357,7 +357,8 @@ func (h *defaultHaTracker) loop(ctx context.Context) error {
 	h.client.WatchPrefix(ctx, "", func(key string, value interface{}) bool {
 		replica, ok := value.(*ReplicaDesc)
 		if !ok {
-			return false
+			level.Error(h.logger).Log("msg", "failed to process value", "return type", fmt.Sprintf("%T", value), "key", key)
+			return true
```

Comment:
```suggestion
			// Always return true to ensure WatchPrefix() is never interrupted, otherwise the HA tracker stops to receive updates.
			return true
```